src/main/java/io/cryostat/recordings/RecordingArchiveHelper.java

```diff
@@ -121,8 +138,50 @@ public void deleteRecording(ConnectionDescriptor connectionDescriptor, String re
                 });
     }
 
+    public List<Map<String, String>> getRecordings() throws Exception {
+        if (!fs.exists(recordingsPath)) {
+            throw new ArchivePathException(recordingsPath.toString(), "does not exist");
+        }
+        if (!fs.isReadable(recordingsPath)) {
+            throw new ArchivePathException(recordingsPath.toString(), "is not readable");
+        }
+        if (!fs.isDirectory(recordingsPath)) {
+            throw new ArchivePathException(recordingsPath.toString(), "is not a directory");
+        }
+        WebServer webServer = webServerProvider.get();
+        List<String> files = this.fs.listDirectoryChildren(recordingsPath);
+        return files.stream()
+                .map(
+                        file -> {
+                            String serviceUriHash = file.split("/")[0];
+                            String recordingName = file.split("/")[1];
+                            try {
+                                return Map.of(
+                                        "serviceUriHash",
+                                        serviceUriHash,
+                                        "name",
+                                        recordingName,
+                                        "reportUrl",
+                                        webServer.getArchivedReportURL(recordingName),
+                                        "downloadUrl",
+                                        webServer.getArchivedDownloadURL(recordingName));
+                            } catch (SocketException
+                                    | UnknownHostException
+                                    | URISyntaxException e) {
+                                logger.warn(e);
+                                return null;
+                            }
+                        })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
     private String writeRecordingToDestination(
             JFRConnection connection, IRecordingDescriptor descriptor) throws Exception {
+        URI serviceUri = URIUtil.convert(connection.getJMXURL());
+        Path specificRecordingsPath =
+                recordingsPath.resolve(String.format("%d", serviceUri.hashCode()));
```

Comment:
`String.format("%d")` seems a little overkill here - `Integer.toString()` might be a better fit.

Reply:
Or, we could use something other than the direct `int hashCode()` of the serviceUri. It could instead be the base64 encoding of the serviceUri, for example, which is already naturally a String and contains only characters which are legal for filenames/directories, and as an added bonus can be decoded back into a serviceUri (unlike the hashcode, where that information is lost).

Reply:
I'll switch to base64 encoding then, seems to have more pros then the direct hashing, especially the decoding.

Reply:
https://stackoverflow.com/questions/3945541/base64-encoding-safe-for-filenames

Modified Base64 might be okay, but this discussion makes a good point about using ex. Base32 instead.